saito-core/src/core/data/merkle.rs

```diff
@@ -2,119 +2,200 @@ use crate::common::defs::SaitoHash;
 use crate::core::data::crypto::hash;
 use crate::core::data::transaction::Transaction;
 use rayon::prelude::*;
+use std::collections::LinkedList;
 
 //
 // MerkleTreeLayer is a short implementation that uses the default
 // Saito hashing algorithm. It is also written to take advantage of
 // rayon parallelization.
 //
-#[derive(PartialEq, Debug, Clone)]
+//#[derive(PartialEq, Debug, Clone)]
+
 pub struct MerkleTreeNode {
-    left: SaitoHash,
-    right: SaitoHash,
-    hash: SaitoHash,
+    left: Option<Box<MerkleTreeNode>>,
```

Comment:
why do we need Box?

Reply:
1. Rather than used to generate the root hash, the entire tree will be kept with the block
2. It may contain hundreds of thousands of transactions ?